lib/ansible/modules/network/nxos/nxos_bfd_global.py

```diff
@@ -0,0 +1,308 @@
+    }]
+    output = None
+    try:
+        output = run_commands(module, cmds)[0]
+    except ConnectionError as exc:
+        if 'CLI command error' in repr(exc):
+            output = repr(exc)
+        else:
+            raise
+    return output
+
+
+def get_existing(module, cmd_ref, show_cmd):
+    '''
+    Get a list of existing command states from device; then update cmd_ref
+    with any 'existing' values that differ from default states
+    '''
+    output = execute_show_command(module, show_cmd, 'text') or []
+    if not output:
+        return {}
+
+    # Walk each cmd in cmd_ref, use cmd pattern to discover existing cmds
+    output = output.split('\n')
+    for k in cmd_ref:
+        pattern = cmd_ref[k]['getval']
+        match = [m.groups() for m in (re.search(pattern, line) for line in output) if m]
+        if not match:
+            continue
+        match = list(match[0]) # tuple to list
+        # handle 'no' keyword
+        if None is match[0]:
+            match.pop(0)
+        elif 'no' in match[0]:
+            cmd_ref[k]['no_cmd'] = True
+            match.pop(0)
+            if not match:
+                continue
+        type = cmd_ref[k]['type']
+        if 'int' == type:
+            cmd_ref[k]['existing'] = int(match[0])
+        if 'int_list' == type:
+            cmd_ref[k]['existing'] = [int(i) for i in match]
+        if 'str' == type:
+            cmd_ref[k]['existing'] = match[0]
+
+    logit(cmd_ref)
+    return cmd_ref
+
+
+def get_playvals(module, cmd_ref):
+    '''Update cmd_ref with playbook values
+    '''
+    for k in cmd_ref.keys():
+        if k in module.params and module.params[k] is not None:
+            playval = module.params[k]
+            if 'int' is cmd_ref[k]['type']:
+                playval = int(playval)
+            elif 'int_list' is cmd_ref[k]['type']:
+                playval = [int(i) for i in playval]
+            cmd_ref[k]['playval'] = playval
+
+    return cmd_ref
+
+
+def get_proposed(cmd_ref):
+    '''
+    Compare playbook values against existing states and create a list
+    of cli commands to play on the device.
+    '''
+    proposed = []
+    # Create a list of playbook values
+    playvals = [k for k,v in cmd_ref.items() if 'playval' in v]
+    # Compare against current state
+    for k in playvals:
+        playval = cmd_ref[k]['playval']
+        existing = cmd_ref[k].get('existing', cmd_ref[k]['default'])
+        logit('pv: %s ex: %s' %(playval,existing))
+        if playval == existing:
+            continue
+        cmd = None
+        type = cmd_ref[k]['type']
+        if 'int' == type:
+            cmd = cmd_ref[k]['setval'].format(playval)
+        elif 'int_list' == type:
+            cmd = cmd_ref[k]['setval'].format(*(playval))
+        elif 'str' == type:
+            if playval:
+                cmd = cmd_ref[k]['setval'].format('', playval)
+            elif existing:
+                cmd = cmd_ref[k]['setval'].format('no ', existing)
```

Comment:
Here also, possibly an 'else' for an unknown type?

Reply:
Agreed.